src/theme.c

```diff
@@ -56,6 +57,62 @@ drop(struct lab_data_buffer **buffer)
 	}
 }
 
+static cairo_surface_t *
+get_surface(struct lab_data_buffer *buffer)
+{
+	/* Handle CAIRO_FORMAT_ARGB32 buffers */
+	if (buffer->cairo) {
+		return cairo_get_target(buffer->cairo);
+	}
+
+	/* Handle DRM_FORMAT_ARGB8888 buffers */
+	int w = buffer->unscaled_width;
+	int h = buffer->unscaled_height;
+	cairo_surface_t *surface =
+		cairo_image_surface_create(CAIRO_FORMAT_ARGB32, w, h);
+	if (!surface) {
+		return NULL;
+	}
+	unsigned char *data = cairo_image_surface_get_data(surface);
+	cairo_surface_flush(surface);
+	memcpy(data, buffer->data, h * buffer->stride);
+	cairo_surface_mark_dirty(surface);
+	return surface;
+}
+
+static void
+create_hover_fallback(struct theme *theme, struct lab_data_buffer **hover_buffer,
+		struct lab_data_buffer *icon_buffer)
+{
+	assert(icon_buffer);
+	assert(!*hover_buffer);
+
+	cairo_surface_t *icon = get_surface(icon_buffer);
```

Comment:
This may leak the cairo surface on `Reconfigure`. We need some way to `cairo_surface_destroy(icon)` but only when it was actually created in `get_surface()` (the `DRM_FORMAT_ARGB8888`) case.
Maybe something like
```c
struct surface_wrap {
	bool free_surface;
	cairo_surface_t *surface;
};

static struct surface_wrap
get_surface()
{
	..
}

static void
create_hover_fallback(..)
{
	..
	if (icon->free_surface) {
		cairo_surface_destroy(icon->surface);
	}
}
```
?